provider/transformation.go

```diff
@@ -0,0 +1,209 @@
+package provider
+
+import (
+	"fmt"
+)
+
+type (
+	OutTransformationProvider[T any] interface {
+		convertToInt(T) (int64, error)
+		convertToString(T) (string, error)
+		convertToFloat(T) (float64, error)
+		convertToBool(T) (bool, error)
+		outTransformations() []OutTransformation
+	}
+	InTransformationProvider interface {
+		setParam(string, any) error
+		inTransformations() []InTransformation
+	}
+	ScriptTransformationProvider[A any] interface {
+		evaluate() (A, error)
+	}
+)
+
+type TransformationConfig struct {
+	Name, Type string
+	Config     Config
+}
+
+type InTransformation struct {
+	name     string
+	function func() (any, error)
+}
+
+type OutTransformation struct {
+	name, Type string
+	function   func(any) error
+}
+
+func ConvertInFunctions(inConfig []TransformationConfig) ([]InTransformation, error) {
+	var in []InTransformation
+
+	for _, cc := range inConfig {
+		var f func() (any, error)
+
+		switch cc.Type {
+		case "bool":
+			ff, err := NewBoolGetterFromConfig(cc.Config)
+			if err != nil {
+				return nil, fmt.Errorf("%s: %w", cc.Name, err)
+			}
+			f = func() (any, error) { return ff() }
+
+		case "int":
+			ff, err := NewIntGetterFromConfig(cc.Config)
+			if err != nil {
+				return nil, fmt.Errorf("%s: %w", cc.Name, err)
+			}
+			f = func() (any, error) { return ff() }
+
+		case "float":
+			ff, err := NewFloatGetterFromConfig(cc.Config)
+			if err != nil {
+				return nil, fmt.Errorf("%s: %w", cc.Name, err)
+			}
+			f = func() (any, error) { return ff() }
+
+		case "string":
+			ff, err := NewStringGetterFromConfig(cc.Config)
+			if err != nil {
+				return nil, fmt.Errorf("%s: %w", cc.Name, err)
+			}
+			f = func() (any, error) { return ff() }
+
+		default:
+			return nil, fmt.Errorf("%s: Could not find converter for %s", cc.Name, cc.Type)
+		}
+
+		in = append(in, InTransformation{
+			name:     cc.Name,
+			function: f,
+		})
+	}
+	return in, nil
+}
+
+func ConvertOutFunctions(outConfig []TransformationConfig) ([]OutTransformation, error) {
+	var out []OutTransformation
+
+	for _, cc := range outConfig {
+		var f func(v any) error
+
+		switch cc.Type {
+		case "bool":
+			ff, err := NewBoolSetterFromConfig(cc.Name, cc.Config)
+			if err != nil {
+				return nil, fmt.Errorf("%s: %w", cc.Name, err)
+			}
+
+			f = func(v any) error {
+				b, ok := v.(bool)
+				if !ok {
+					return fmt.Errorf("%s: could not convert %v to bool", cc.Name, b)
```

Comment:
Can this ever happen unless we're making a programming error? if not we could force-cast it without the check and have it panic, but I'm not sure.

Reply:
I would agree that this can only be caused by a programming error. The `convertToXXX` functions should have taken care of the conversation. The type cast is just an artifact here, since we loose the type information on the way to this function. I have replaced this with force-casts